Source/SwiftLintFramework/Extensions/File+SwiftLint.swift

```diff
@@ -181,4 +181,32 @@ extension File {
         }
         return violatingRanges
     }
+
+    internal func numberOfCommentAndWhitespaceOnlyLines(startLine: Int, endLine: Int) -> Int {
+        let commentKinds = Set(SyntaxKind.commentKinds())
+
+        return syntaxKindsByLines.filter { line, kinds -> Bool in
+            guard line >= startLine && line <= endLine else {
+                return false
+            }
+
+            // if the line has only whitespace, `kinds` will be an empty array
+            return kinds.filter { !commentKinds.contains($0) }.isEmpty
+        }.count
+    }
+
+    internal func lineCount(startLine: Int, endLine: Int) -> Int {
+        let ignoredLines = numberOfCommentAndWhitespaceOnlyLines(startLine, endLine: endLine)
+        return endLine - startLine - ignoredLines
+    }
+
+    internal func exceedsLineCountExcludingCommentsAndWhitespace(start: Int, _ end: Int,
+                                                                 _ limit: Int) -> (Bool, Int) {
+        if end - start <= limit {
+            return (false, end - start)
+        }
+
+        let count = lineCount(start, endLine: end)
```

Comment:
Removing the `lineCount()` function and having the following inline would help improve code locality:

`let count = end - start - numberOfCommentAndWhitespaceOnlyLines(start, endLine: end)`